clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesStatefulSetCachingAgent.java

```diff
@@ -76,7 +77,13 @@ protected KubernetesKind primaryKind() {
     Map<KubernetesManifest, List<KubernetesManifest>> result = new HashMap<>();
 
     for (KubernetesManifest manifest : primaryResourceList) {
-      result.put(manifest, Collections.singletonList(services.get(KubernetesStatefulSetHandler.serviceName(manifest))));
+      String serviceName = KubernetesStatefulSetHandler.serviceName(manifest);
```

Comment:
SGTM

Reply:
Just kidding... it's kind of ugly because the start & end of the loop rely on the same variable (`manifest`). If java had a neater way of expressing tuples it'd be easy to wrap together a service + manifest pair and do a map -> filter -> map -> collectBy, but doing it here just becomes ugly